_posts/2020-01-17-strong-parameters-migration-guide.markdown

```diff
@@ -0,0 +1,97 @@
+---
+layout: post
+title: "The Complete Guide to Migrate to Strong Parameters"
+date: 2020-01-21 13:00:00
+categories: ["rails", "upgrades"]
+author: luciano
+---
+
+Migrating from `Protected Attributes` to `Strong Parameters` in a [Rails](https://rubyonrails.org/) project can be a huge step of the upgrade process. Specially when we are [upgrading a large application](https://www.fastruby.io/blog/rails/upgrades/case-study/upgrading-a-large-rails-application-from-4.2-to-5.2.html). This guide is meant to help you tackle that step faster and with a lot less pain.
+
+<!--more-->
+
+### Protected Attributes & Strong Parameters
+
+To give you a bit of context, let's recap what `Protected Attributes` and [Strong Parameters](https://guides.rubyonrails.org/action_controller_overview.html#strong-parameters) actually are. They are two different Rails implementations for protecting attributes from end-user injection (a.k.a. [Mass Assignment](https://owasp.org/www-project-cheat-sheets/cheatsheets/Mass_Assignment_Cheat_Sheet.html))
+
+To understand what are the benefits of `Strong Parameters` over `Protected Attributes`, I recommend you checking [this RailsCasts episode](http://railscasts.com/episodes/371-strong-parameters?autoplay=true).
+
+`Protected Attributes` was part of the Rails core since the beginning of it until Rails 3.2. In Rails 4.0 they introduced `Strong Parameters` as a replacement of it, and it has been part of the core since then. Before that it was possible to use it through the [strong_parameters](https://github.com/rails/strong_parameters) gem.
+
+After Rails 4.0 came out with the implementation of `Strong Parameters`, we were able to have backwards compatibility with `Protected Attributes` by using the [protected_attributes](https://github.com/rails/protected_attributes) gem. That way you could have the latest version of Rails without migrating to `Strong Parameters`.
+
+That gem was supported by the Rails team until the release of Rails 5.0. After that we were forced to migrate to `Strong Parameters`. Once that happened we started to get unofficial forks of `protected_attributes` that support the latest version of Rails, like [protected_attributes_continued](https://github.com/westonganger/protected_attributes_continued).
+
+At this point we strongly recommend to fully migrate to `Strong Parameters`, since the available options to keep `Protected Attributes` alive, have very limited support and can encounter security issues.
```

Comment:
```suggestion
At this point we strongly recommend fully migrating to `Strong Parameters`, since the available options to keep `Protected Attributes` alive have very limited support and can encounter security issues.
```